aten/src/ATen/native/MaxUnpooling.cpp

```diff
@@ -14,16 +14,16 @@ Tensor& max_unpooling2d_forward_out_cpu(
   auto owidth = output_size[1];
   TORCH_CHECK(
       indices_.scalar_type() == at::ScalarType::Long,
-      "elements in indices should be type int64");
+      "elements in indices should be type int64 but got: ", indices_.scalar_type());
   TORCH_CHECK(
       output_size.size() == 2,
-      "There should be exactly two elements (height, width) in output_size");
+      "There should be exactly two elements (height, width) in output_size, but got ", output_size.size(), " elements.");
   TORCH_CHECK(
       (self_.ndimension() == 3 || self_.ndimension() == 4),
-      "Input to max_unpooling2d should be a 3d or 4d Tensor");
+      "Input to max_unpooling2d should be a 3d or 4d Tensor, but got tensor with dimension: ", self_.ndimension());
   TORCH_CHECK(
       self_.sizes() == indices_.sizes(),
-      "Shape of indices should match shape of input");
+      "Shape of indices (", indices_.sizes(), ") should match shape of input (", self_.sizes(), ")");
```

Comment:
Moving `indices_.sizes()` to the end of this message with a "but got" would improve consistency with other error messages and may improve readability. Presently, this error mesage reverses the order of the "expected" value and the "actual" value when compared to other error messages.

Reply:
This makes sense, consistency in error messages is always great to have, I agree! Thanks for the suggestion, @saketh-are - I've made the requested changes, hope they look good.